pkg/logger/ecs_error_log.go

```diff
@@ -0,0 +1,175 @@
+package logger
+
+import (
+	"fmt"
+	"go.uber.org/zap/zapcore"
+	"os"
+	"path/filepath"
+	"strconv"
+	"time"
+)
+
+var ErrorFilePath = os.Getenv("STATUS_MESSAGE_FILE_PATH")
+
+var errorLogMaxAge = os.Getenv("STATUS_MESSAGE_TTL")
+var errorLogMaxAgeInHours, _ = strconv.Atoi(errorLogMaxAge)
+
+var size = os.Getenv("STATUS_MESSAGE_MAX_BYTE_LENGTH")
+var errorFileSize, _ = strconv.Atoi(size)
```

Comment:
we need to be able to handle errors in this code, including failure to convert the strings to ints.

I think this code may be should be added inside of a function.

What if you had a structure to store all of these:

```
type ECSErrorLogger struct {
ErrorFilePath string
errorLogMaxAge string
// put all of the error config options here
}

// then there is a create method which gets the env vars and sets everything:
func createECSErrorLogger() ECSErrorLogger {
os.getenv...
// set up the struct and return it
}
```

And then your other functions are methods in the structure:

```
// the methods can access the member fields on ECSErrorLogger structure
func (logger ECSErrorLogger) ECSErrorReporter() {}
func (logger ECSErrorLogger) WriteError() {}
```

Reply:
@pingleig @gavindoudou What do you think of this approach?

Reply:
If you go with this approach, move all of the global variables to that structure if possible. I don't like global variables. I think wrapping the behavior in a structure is better organization and cleaner.

Reply:
Agree, that will increase the readability and make the code more organized.

Reply:
i'm ok with this change

Reply:
Nice explanation! Very clear structure. Agree